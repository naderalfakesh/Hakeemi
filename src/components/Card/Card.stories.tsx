import React, { Fragment } from 'react';
import { View, StyleSheet } from 'react-native';
import { storiesOf } from '@storybook/react-native';
import Card from './Card';

const Container = (storyFn: any) => (
  <View style={styles.container}>{storyFn()}</View>
);

const CardDemo = () => (
  <Fragment>
    <Card
      title="This is title"
      subtitle="Lorem ipsum dolor sit amet consectetur adipisicing elit. Cumque, sit!"
    />
  </Fragment>
);

storiesOf('Card', module)
  .addDecorator(Container)
  .add('default', () => <CardDemo />);

const styles = StyleSheet.create({
  container: {
    flexGrow: 1,
    padding: 8,
    justifyContent: 'space-evenly',
  },
});
