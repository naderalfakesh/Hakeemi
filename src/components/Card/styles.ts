import { StyleSheet } from 'react-native';
import colors from '../../theme/colors';
import spacing from '../../theme/spacing';

export const ICON_SIZE = 40;

export default StyleSheet.create({
  base: {
    paddingTop: spacing.medium1,
  },
  card: {
    paddingTop: spacing.small3,
    paddingHorizontal: spacing.medium3,
    backgroundColor: colors.primary,
    borderRadius: 30,
    flexDirection: 'row',
  },
  textContainer: {
    marginVertical: spacing.small3,
    flexShrink: 1,
  },
  rightNode: {},
  icon: {
    position: 'absolute',
    top: 0,
    right: 0,
    zIndex: 1,
    width: ICON_SIZE,
    height: ICON_SIZE,
    borderRadius: ICON_SIZE * 0.5,
    overflow: 'hidden',
    color: colors.danger,
    backgroundColor: 'white',
  },
  opacity: { opacity: 0.7 },
});
