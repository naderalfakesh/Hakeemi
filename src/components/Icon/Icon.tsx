import React, { FC } from 'react';
import { ColorValue, StyleSheet } from 'react-native';
import { createIconSetFromFontello } from 'react-native-vector-icons';
import { IconProps as BaseIconProps } from 'react-native-vector-icons/Icon';
import * as config from './config.json';
const BaseIcon = createIconSetFromFontello(config);

export type IconName =
  | 'ambulance'
  | 'diagnose'
  | 'nav-menu'
  | 'heart-plus'
  | 'badge'
  | 'heart-beat'
  | 'pill'
  | 'heart'
  | 'checkup'
  | 'heart-fill'
  | 'search'
  | 'cancel'
  | 'close-filled'
  | 'left-arrow'
  | 'shot';

export interface IconProps extends Omit<BaseIconProps, 'color'> {
  name: IconName;
  color?: ColorValue;
}

const Icon: FC<IconProps> = ({ style, size = 24, color, ...otherProps }) => (
  <BaseIcon
    style={StyleSheet.compose({ lineHeight: size, textAlign: 'center' }, style)}
    size={size}
    color={color as string}
    {...otherProps}
  />
);

export default Icon;
