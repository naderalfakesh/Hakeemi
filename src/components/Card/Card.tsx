import React, { FC, ReactNode } from 'react';
import { View, ViewStyle } from 'react-native';
import Icon, { IconName } from '../Icon';
import Text from '../Text';
import styles, { ICON_SIZE } from './styles';

interface Props {
  style?: ViewStyle;
  containerStyle?: ViewStyle;
  title: string;
  subtitle?: string;
  rightNode?: ReactNode;
  closeIconName?: IconName;
  onClose?: () => void;
}

const Card: FC<Props> = ({
  style,
  containerStyle,
  title,
  subtitle,
  rightNode,
  closeIconName,
  onClose,
}) => {
  return (
    <View testID="card-container" style={[styles.base, containerStyle]}>
      <Icon
        testID="close-icon"
        name={closeIconName ? closeIconName : 'close-filled'}
        style={styles.icon}
        size={ICON_SIZE}
        onPress={onClose}
      />
      <View testID="card" style={[styles.card, style]}>
        <View style={styles.textContainer}>
          <Text color="white" level="600" size="head-24">
            {title}
          </Text>
          <Text color="white" size="body-14" style={styles.opacity}>
            {subtitle ? subtitle : ''}
          </Text>
        </View>
        {rightNode ? (
          <View style={styles.rightNode}>{rightNode}</View>
        ) : undefined}
      </View>
    </View>
  );
};

export default Card;
