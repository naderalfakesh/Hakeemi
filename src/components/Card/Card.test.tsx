import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Card from './Card';
import Icon from '../Icon';
import { StyleSheet } from 'react-native';

describe('Button component test', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <Card
        title="Card title"
        subtitle="Card subtitle"
        rightNode={<Icon name="heart-plus" />}
        closeIconName="checkup"
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('overrides styles', () => {
    const { getByTestId } = render(
      <Card
        containerStyle={{ padding: 0 }}
        style={{ padding: 0 }}
        title="Card title"
      />,
    );

    const cardContainer = getByTestId('card-container');
    expect(StyleSheet.flatten(cardContainer.props.style).padding).toBe(0);

    const card = getByTestId('card');
    expect(StyleSheet.flatten(card.props.style).padding).toBe(0);
  });

  it('invokes a function when close icon is pressed', () => {
    const mockFn = jest.fn();
    const { getByTestId } = render(
      <Card title="Card title" onClose={mockFn} />,
    );
    const closeIcon = getByTestId('close-icon');
    fireEvent.press(closeIcon);
    expect(mockFn).toBeCalled();
  });
});
