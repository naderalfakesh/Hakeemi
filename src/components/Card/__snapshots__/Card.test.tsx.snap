// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingTop": 20,
      },
      undefined,
    ]
  }
  testID="card-container"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 40,
        },
        Array [
          Object {
            "lineHeight": 40,
            "textAlign": "center",
          },
          Object {
            "backgroundColor": "white",
            "borderRadius": 20,
            "color": "rgba(255, 87, 34, 1)",
            "height": 40,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 40,
            "zIndex": 1,
          },
        ],
        Object {
          "fontFamily": "doctor-font",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    testID="close-icon"
  >
    
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 188, 212, 1)",
          "borderRadius": 30,
          "flexDirection": "row",
          "paddingHorizontal": 32,
          "paddingTop": 16,
        },
        undefined,
      ]
    }
    testID="card"
  >
    <View
      style={
        Object {
          "flexShrink": 1,
          "marginVertical": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontFamily": "System",
            "fontSize": 24,
            "fontWeight": "600",
            "includeFontPadding": false,
            "lineHeight": 30,
            "textAlign": undefined,
          }
        }
        testID="Text"
      >
        Card title
      </Text>
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "400",
            "includeFontPadding": false,
            "lineHeight": 18,
            "opacity": 0.7,
            "textAlign": undefined,
          }
        }
        testID="Text"
      >
        Card subtitle
      </Text>
    </View>
    <View
      style={Object {}}
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 24,
            },
            Object {
              "lineHeight": 24,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "doctor-font",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
